File: backend/src/main/java/net/lycee/web/enquete/api/entity/SpaceEntity.java
package net.lycee.web.enquete.api.entity;

import lombok.Data;
import net.lycee.web.enquete.api.domain.SpaceId;
import net.lycee.web.enquete.api.domain.UserId;

import java.time.LocalDateTime;

@Data
public class SpaceEntity {
    private SpaceId id;
    private UserId ownerId;
    private String name;
    private LocalDateTime openedTime;
    private LocalDateTime closeTime;

    public static SpaceEntity of(
            SpaceId id,
            UserId ownerId,
            String name,
            LocalDateTime openedTime,
            LocalDateTime closeTime
    ) {
        SpaceEntity entity = new SpaceEntity();
        entity.setId(id);
        entity.setOwnerId(ownerId);
        entity.setName(name);
        entity.setOpenedTime(openedTime);
        entity.setCloseTime(closeTime);
        return entity;
    }
}

File: frontend/src/types/SpaceEntity.ts
import type { SpaceId, UserId } from "./common.ts";
import type { SpaceInfo } from "./api.ts";

export class SpaceEntity {
  // スペースID
  id: SpaceId;
  // 開設ユーザID
  ownerId: UserId;
  // スペース名
  name: string;
  // 開設日時
  openedDate: Date;
  // 終了日時
  closeDate: Date;
  // スペース画像（予約）
  img?: string;
  // 概要（予約）
  description?: string;

  constructor(
    id: SpaceId,
    ownerId: UserId,
    name: string,
    openedDate: Date,
    closeDate: Date,
    roomImg?: string | undefined,
  ) {
    this.id = id;
    this.ownerId = ownerId;
    this.name = name;
    this.openedDate = openedDate;
    this.closeDate = closeDate;
    this.img = roomImg;
  }

  static from({ id, ownerId, name, openedTime, closeTime }: SpaceInfo) {
    return new SpaceEntity(
      id,
      ownerId,
      name,
      new Date(openedTime),
      new Date(closeTime),
    );
  }
}
